### PR TITLE
Fix multi-byte character handling in `read_file`

### PR DESCRIPTION
**Pull request by sidewaysthought**
### Background
Related to issue #232 - Multi-byte character handling
User reports UTF files not being read. Some files may be UTF-8 but have a BOM marker which Python can see as invalid, or may not be UTF-8 at all.

### Changes
The read_file command has been updated to detect the file encoding, then use the detected encoding to provide more reliable reading of files. The change requires the addition of the chardet dependency. It has been added to requirements.txt

### Documentation
Change is minor with what I believe to be obvious intent when viewing the code.

### Test Plan
Tested by making agent create a file by output redirect from PowerShell (which includes BOM markers, and previously failed read.) Files now correctly read.

### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [X] I have thoroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes

Unfamiliar with generating test cases on a project of this scale. Would rather not horribly break things.

### CODE REVIEW

**Comment by SpaceTTAngel** [comment]
For me it's ok, I've tested the file_operations.py and requirements.txt with several datasets @sidewaysthought

**Review by SpaceTTAngel — APPROVED** [review]
I've tested the code with a several files dataset to stimulate encoding reading, with the following logs 👍 

encoded file type: ascii
encoded file content: ciaociao

***
encoded file type: ascii
encoded file content: ciaociao
***
encoded file type: ascii
encoded file content: ciaociao
***
encoded file type: UTF-8-SIG
encoded file content: ciaociao
***
encoded file type: UTF-16
encoded file content: ciaociao
***
encoded file type: UTF-16
encoded file content: ciaociao
***
encoded file type: UTF-16
encoded file content: ciaociao
***
encoded file type: UTF-16
encoded file content: ciaociao
***
encoded file type: UTF-16
encoded file content: ciaociao
***
encoded file type: ascii
encoded file content: premi?re is first
premie?re is slightly different
????????? is Cyrillic
? am Deseret

***
encoded file type: Windows-1254
encoded file content: premiŠre is first
premie?re is slightly different
????????? is Cyrillic
? am Deseret

***
encoded file type: KOI8-R
encoded file content: premi?re is first
premie?re is slightly different
Кириллица is Cyrillic
? am Deseret

***
encoded file type: ISO-8859-1
encoded file content: première is first
premie?re is slightly different
????????? is Cyrillic
? am Deseret

***
encoded file type: Windows-1254

**Comment by ashreim-UPL** [comment]
I had the same issue so I removed the file and replaced it with txt file. But this is definitely a bug

File: requirements.txt [hunk 1]
```diff
@@ -21,6 +21,7 @@ webdriver-manager
 jsonschema
 tweepy
 click
+chardet>=5.1.0
```
Comment (Pwuts) [inline requirements.txt]: Please use `charset-normalizer` instead

**Review by Pwuts — CHANGES_REQUESTED** [review]

**Review by Pwuts — APPROVED** [review]

**Pull request merged**